File: Java/src/com/leetcode/Medium/ValidateStackSequences.java
package com.leetcode.Medium;

import java.util.Stack;

// 946. Validate Stack Sequences
public class ValidateStackSequences {
    public boolean validateStackSequences(int[] pushed, int[] popped) {
        Stack<Integer> stack = new Stack<>();
        int i = 0;
        for (int j : pushed) {
            stack.push(j);
            while (!stack.isEmpty() && stack.peek() == popped[i]) {
                stack.pop();
                i++;
            }
        }

        return i == popped.length;
    }
}

File: JavaScript/Medium/ValidateStackSequences.js
/**
 * 946. Validate Stack Sequences
 * @param {number[]} pushed
 * @param {number[]} popped
 * @return {boolean}
 */
var validateStackSequences = function (pushed, popped) {
  let stack = [];
  let i = 0;
  for (let j of pushed) {
    stack.push(j);
    while (stack.length > 0 && stack[stack.length - 1] === popped[i]) {
      stack.pop();
      i++;
    }
  }

  return i === popped.length;
};
